### PR TITLE
Assertion testing

### PR DESCRIPTION
**Pull request by descorp**
### Changes

* Wrap assertionFailure into AdyenAssertion.assert
* add lunch argument `-UITests` on project level
* apply Xcode recommended settings and get read of warning

### CODE REVIEW

File: Adyen/Utilities/Debugging/Assertion.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import Foundation
+
+/// A typealias for a closure that handles a URL through which the application was opened.
+/// :nodoc:
+public typealias AssertionListner = (String) -> Void
+
+// :nodoc:
+public enum AdyenAssertion {
+
+    public static var listner: AssertionListner?
```
Comment (mohammedDehairy) [inline Adyen/Utilities/Debugging/Assertion.swift:16]: maybe this should be `internal` if its only accessed by unit tests.

File: AdyenActions/Components/3DS2/Action handlers/ThreeDS2CoreActionHandler.swift [hunk 2]
```diff
@@ -110,40 +110,31 @@ internal class ThreeDS2CoreActionHandler: Component {
...
-            let challengeParameters = ADYChallengeParameters(from: token)
-            transaction.performChallenge(with: challengeParameters) { [weak self] challengeResult, error in
-                self?.handle(challengeResult, error: error,
-                             for: challengeAction,
-                             completionHandler: completionHandler)
-            }
+            token = try Coder.decodeBase64(challengeAction.challengeToken) as ThreeDS2Component.ChallengeToken
         } catch {
-            didFail(with: error, completionHandler: completionHandler)
+            return didFail(with: error, completionHandler: completionHandler)
```
Comment (mohammedDehairy) [inline AdyenActions/Components/3DS2/Action handlers/ThreeDS2CoreActionHandler.swift:122]: no need for `return`

File: Tests/AdyenSwiftUITests/AdyenSwiftUITests.swift [hunk 3]
```diff
@@ -15,7 +15,6 @@ class AdyenSwiftUITests: XCTestCase {
     override func setUp() {
         super.setUp()
         app = XCUIApplication()
-        app.launchArguments = ["-UITests"]
```
Comment (mohammedDehairy) [inline Tests/AdyenSwiftUITests/AdyenSwiftUITests.swift:18]: why is this deleted?

File: Adyen/Utilities/Debugging/Assertion.swift (hunk 1 above)
Comment (descorp) [reply Adyen/Utilities/Debugging/Assertion.swift:16]: Good point 🤔

File: Tests/AdyenSwiftUITests/AdyenSwiftUITests.swift (hunk 3 above)
Comment (descorp) [reply Tests/AdyenSwiftUITests/AdyenSwiftUITests.swift:18]: I have injected it on [project level](https://github.com/Adyen/adyen-ios/pull/427/files#diff-74c105d362dd8cd9fc6b8ae1c32803c6134ace094ca58d05464ab88c399cf529R42)

Comment (mohammedDehairy) [reply Tests/AdyenSwiftUITests/AdyenSwiftUITests.swift:18]: 👍

**Review by mohammedDehairy — DISMISSED** [review]

**Review by mohammedDehairy — APPROVED** [review]
